pystiche_papers/sanakoyeu_et_al_2018/_modules.py

```diff
@@ -184,3 +193,87 @@ def residual_block(channels: int) -> ResidualBlock:
     )
 
     return ResidualBlock(residual)
+
+
+def encoder(in_channels: int = 3,) -> SequentialEncoder:
+    r"""Encoder part of the :class:`Transformer` from :cite:`SKL+2018`.
+
+    Args:
+        in_channels: Number of channels in the input. Defaults to ``3``.
+
+    """
+    modules = [
+        nn.ReflectionPad2d(15),
+        ConvBlock(in_channels=in_channels, out_channels=32, kernel_size=3, stride=1),
+    ]
+    channels = (32, 32, 64, 128, 256)
+
+    conv_block = partial(ConvBlock, kernel_size=3, stride=2)
+
+    modules.extend(
+        [
+            conv_block(in_channels, out_channels)
+            for in_channels, out_channels in more_itertools.pairwise(channels)
+        ]
+    )
+
+    return SequentialEncoder(modules)
+
+
+def decoder(
+    out_channels: int = 3, num_residual_blocks: int = 9,
+) -> pystiche.SequentialModule:
+
+    modules: List[nn.Module] = []
+    for i in range(num_residual_blocks):
+        modules.append(residual_block(256))
+
+    channels = (256, 256, 128, 64, 32)
+    upsample_conv_block = partial(UpsampleConvBlock, kernel_size=3)
+    modules.extend(
+        [
+            upsample_conv_block(in_channels=in_channels, out_channels=out_channels)
+            for in_channels, out_channels in more_itertools.pairwise(channels)
+        ]
+    )
+    modules.append(nn.ReflectionPad2d(3))
+    modules.append(
+        conv(
+            in_channels=32,
+            out_channels=out_channels,
+            kernel_size=7,
+            stride=1,
+            padding="valid",
+        )
+    )
+    return pystiche.SequentialModule(*modules)
+
+
+class Decoder(nn.Module):
```

Comment:
This is needed for the properties handling.

```suggestion
class Decoder(pystiche.Module):
```